docs/library: Add machine.DAC documentation.

mattytrentini opened the pull request:
### Summary

As described in #7915, `machine` module needed a `DAC` entry. 

This turned out to be a simple doc item but was surprisingly challenging to summarise the different DAC implementations.

### Testing

Tested by building the docs locally; looks reasonable.

### Trade-offs and Alternatives

It is tempting to overhaul the DAC interface to make it more complete and remove some of the differences between ports.

For example, we could settle on using, say, a 16 bit range for all ports and interpolate with however many bits are available. Something like a `DAC.bits()` method could return the port-specific number of bits so the user could determine the actual resolution available. The docs would not need to call out the port-specific ranges. Also, `init` and `deinit` are not standard, they probably should be.

So, rather than overhaul the interface (and introduce breaking changes), I thought it best to document the current state and then, later, make the changes and update the docs.

mattytrentini committed:
docs/library: Add machine.DAC documentation.

Fixes issue #7915.

Signed-off-by: Matt Trentini <matt.trentini@gmail.com>

robert-hh commented:
Thank you for the effort. Minor comments:

- How about mentioning the STM32's pyb.DAC() module.
- Given that there is not any more that creating the DAC instance, DAC.init() would not have any functionality. And a DAC.deinit() would only be needed if the DAC implementation would allocate resources which may be used by other modules. I did not see that.

mattytrentini commented on docs/library/machine.DAC.rst:
```diff
@@ -0,0 +1,53 @@
+Constructors
+------------
+
+.. class:: DAC(pin)
```
Ooof, ok, thanks. I can see why you did it that way but it's amazing how - across just three ports - a single method can differ so much! 😉

mattytrentini commented:
Updated:
- Included STM32 (and mentioned DAC was implemented via `pyb.DAC`)
- Referenced ESP32 and SAMD Quickref's where there is a little more detail
- Updated constructor to accept an 'id' that can be an index or a Pin object

robert-hh commented on docs/library/machine.DAC.rst:
```diff
@@ -0,0 +1,66 @@
+
+  Construct a new DAC object.
+
+  ``id`` is a pin object (ESP32 and Renesas RA) or an index to a DAC resource
```
1. There is no machine.DAC for STM32. It's pyb.DAC, and it has more features like write_timed.
2. On ESP32 and Renesas and STM32 `id` must be a value that refers to a pin. It may be a Pin object, but can as well be a number or string (Renesas, STM32), as supported by the port.

mattytrentini replied:
Updated to remove any mention of stm32 except for a note that refers to 'similar functionality' in `pyb,DAC`.

dpgeorge replied:
Using "pin object" is consistent with most of the other parts of the docs, including `machine.ADC`.  So I think it's fine to use here as well.
